shared/util/tf/tf_parser.go

```diff
@@ -103,3 +110,129 @@ func (tf TfParser) ParseServices(dir string) (map[string]bool, error) {
 	}
 	return services, nil
 }
+
+func (tf TfParser) ParseModuleCall(dir string) (ModuleCall, error) {
+	moduleCall := ModuleCall{Parameters: map[string]any{}}
+
+	err := filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() {
+			if d.Name() == ".terraform" || d.Name() == ".git" {
+				return filepath.SkipDir
+			}
+			return nil
+		}
+		if filepath.Ext(path) != ".tf" {
+			return nil
+		}
+
+		b, err := os.ReadFile(path)
+		if err != nil {
+			return err
+		}
+
+		f, diags := hclsyntax.ParseConfig(b, path, hcl.Pos{Line: 1, Column: 1})
+		if diags.HasErrors() {
+			return errors.Wrapf(diags.Errs()[0], "failed to parse %s", path)
+		}
+
+		content, _, diags := f.Body.PartialContent(moduleBlockSchema)
+		if diags.HasErrors() {
+			return errors.New("Terraform code has errors")
+		}
+
+		for _, block := range content.Blocks {
+			if block.Type != "module" {
+				continue
+			}
+
+			attrs, diags := block.Body.JustAttributes()
+			if diags.HasErrors() {
+				return errors.New("Terraform code has errors")
+			}
+			var sourceAttr *hcl.Attribute
+			var ok bool
+			if sourceAttr, ok = attrs["source"]; !ok {
+				// Module without a source
+				continue
+			}
+
+			source, diags := sourceAttr.Expr.(*hclsyntax.TemplateExpr).Parts[0].Value(nil)
+			if diags.HasErrors() {
+				return errors.New("Terraform code has errors")
+			}
+
+			if !strings.Contains(source.AsString(), "modules/happy-stack-") {
+				// Not a happy stack module
+				continue
+			}
+
+			for _, attr := range attrs {
+				value, diag := attr.Expr.Value(nil)
+				if diag.HasErrors() {
+					log.Warnf("Attribute %s cannot be read properly: %s", attr.Name, diag.Errs()[0].Error())
+					//continue
+				}
+
+				v, err := decodeValue(value)
+				if err != nil {
+					continue
+				}
+				if v != nil {
+					moduleCall.Parameters[attr.Name] = v
+				}
+			}
+			delete(moduleCall.Parameters, "image_tag")
+			delete(moduleCall.Parameters, "image_tags")
+			delete(moduleCall.Parameters, "k8s_namespace")
+			delete(moduleCall.Parameters, "stack_name")
+			delete(moduleCall.Parameters, "deployment_stage")
+		}
+
+		return nil
+	})
+	if err != nil {
+		return moduleCall, errors.Wrap(err, "failed to parse terraform files")
+	}
+	return moduleCall, nil
+}
+
+func decodeValue(ctyValue cty.Value) (any, error) {
```

Comment:
How does this handle when the attr is set to a var or a local? (ie `aws_role = var.aws_role`)

Reply:
It will error out on such value. Currently any functional calls or variable references are not supported.